### PR TITLE
HALF ASSED REFACTOR

### PR DESCRIPTION
**Pull request by danyisill**
DIDNT TEST

### CODE REVIEW

**Comment by Demonstrandum** [comment]
yikes...

Time for a..............

CODE :clap: REVIEW :clap: !!!

File: lib/main.ts [hunk 1]
```diff
@@ -52,636 +52,565 @@ const SECRETS = JSON.parse(shell('sh ./generate_secrets.sh').toString());
...
+			// This means the alias has expanded to more than just one word.
+			command = expanded_command_words.shift();
+			expanded_command_words.each(e => args.unshift(e));
+		}
+
+		command = command.toLowerCase();
+		console.log('Received command:', [command, args]);
+
+		var functions = readdirSync('funcs').map(n => n.slice(0, -3))
+		require('./funcs/' + functions[command] + '.js')(message, args, SECRETS, CONFIG)
```
Comment (Demonstrandum) [inline lib/main.ts]: That's going to fail on most commands, you should check if the file exists first, no?

Comment (Demonstrandum) [reply lib/main.ts]:
```js
if (command in functions) {
	// Then require.
}
```

File: lib/main.ts [hunk 2]
```diff
@@ -52,636 +52,565 @@ const SECRETS = JSON.parse(shell('sh ./generate_secrets.sh').toString());
...
+	process_command(message : Message) {
+		const last_command = this._COMMAND_HISTORY.last();
+		this._COMMAND_HISTORY.push(message);
+		if (this._COMMAND_HISTORY.length > CONFIG.commands.max_history) {
+			this._COMMAND_HISTORY.shift();
+		}
+		const current_command = this._COMMAND_HISTORY.last();
+
+		// Try and slow the fellas down a little.
+		if (last_command.channel === current_command.channel) {
```
Comment (Demonstrandum) [inline lib/main.ts:112]: This line and a couple below it were changed a few minutes ago, you wanna pull that quick?

**Review by Demonstrandum — APPROVED** [review]

File: lib/main.ts (hunk 2 above)
Comment (danyisill) [reply lib/main.ts:112]: No idk how to undo it

File: lib/main.ts (hunk 1 above)
Comment (danyisill) [reply lib/main.ts]: functions[command] || 'fallback' innit

**Comment by Demonstrandum** [comment]
Danny, I pulled your changes and have pushed them to main!  Thank you!
If you see anything else that bugs you, just send another :smile: !!

**Pull request closed without merging**